sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala

```diff
@@ -512,7 +511,7 @@ class SQLQueryTestSuite extends QueryTest with SharedSparkSession {
     val schema = df.schema.catalogString
     // Get answer, but also get rid of the #1234 expression ids that show up in explain plans
     val answer = SQLExecution.withNewExecutionId(df.queryExecution, Some(sql)) {
-      hiveResultString(df.queryExecution.executedPlan).map(replaceNotIncludedMsg)
+      hiveResultString(df).map(replaceNotIncludedMsg)
```

Comment:
in `ThriftServerQueryTestSuite`, we get the result by JDBC, so there is no DataFrame created.

We should follow pgsql and return java 8 datetime when the config is enabled. https://jdbc.postgresql.org/documentation/head/8-date-time.html

cc @wangyum @yaooqinn

Reply:
We do not have our own JDBC implementation yet, and the hive JDBC haven't supported this too
https://github.com/apache/hive/blob/ebd41526b9433a196ae0f934bc466cb5c155a202/jdbc/src/java/org/apache/hive/jdbc/HiveQueryResultSet.java#L402-L411